### PR TITLE
Convert Media Control Card to Typescript

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change
Updates Media Control Card to Typescript

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
image
image

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

### CODE REVIEW

**Comment by iantrich** [comment]
hui-legacy-wrapper-card can be removed

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+import "../../../components/ha-card";
+import "../components/hui-warning";
+import "../components/hui-unavailable";
+import "../components/hui-image";
+
+import { applyThemesOnElement } from "../../../common/dom/apply_themes_on_element";
+import { computeStateName } from "../../../common/entity/compute_state_name";
+import { supportsFeature } from "../../../common/entity/supports-feature";
+import { OFF_STATES, SUPPORT_PAUSE } from "../../../data/media-player";
+
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: remove empty line

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+                ? ""
+                : this._computeMediaTitle(stateObj)}
+            </div>
+          </div>
+        </div>
+        ${OFF_STATES.includes(stateObj.state)
+          ? ""
+          : html`
+              <paper-progress
+                max="${stateObj.attributes.media_duration}"
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-media-control-card.ts]:
```suggestion
                max=${stateObj.attributes.media_duration}
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+                value="${stateObj.attributes.media_position}"
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-media-control-card.ts]:
```suggestion
                .value=${stateObj.attributes.media_position}
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+                class="progress"
+              ></paper-progress>
+            `}
+        <div class="controls flex">
+          <div class="left flex">
+            <paper-icon-button
+              icon="hass:power"
+              .action="${stateObj.state === "off" ? "turn_on" : "turn_off"}"
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-media-control-card.ts]:
```suggestion
              .action=${stateObj.state === "off" ? "turn_on" : "turn_off"}
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+          </div>
+          <div class="center flex">
+            <paper-icon-button
+              icon="hass:skip-previous"
+              .action=${"media_previous_track"}
+              @click=${this._handleClick}
+            ></paper-icon-button>
+            <paper-icon-button
+              class="playPauseButton"
+              icon=${this._computeControlIcon(stateObj)}
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: inline this functionality

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+  TemplateResult,
+  customElement,
+  property,
+  css,
+  CSSResult,
+} from "lit-element";
+import { classMap } from "lit-html/directives/class-map";
+import { HassEntity } from "home-assistant-js-websocket";
+import "@polymer/paper-icon-button/paper-icon-button";
+import "@thomasloven/round-slider";
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: unused

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+
+import "../../../components/ha-card";
+import "../components/hui-warning";
+import "../components/hui-unavailable";
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: unused

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+import "../components/hui-image";
```
Comment (iantrich) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: unused

**Review by balloob — APPROVED** [review]
🎉

**Pull request merged**